src/upgrades.ts

```diff
@@ -142,37 +153,46 @@ async function upgradeProxy<T extends Contract>(
 
   // Let the transaction propagate across the ethereum nodes. This is mostly to
   // wait for all Alchemy nodes to catch up their state.
-  await newContractInstance.deployTransaction.wait(1)
+  await newContractInstance.deploymentTransaction()?.wait(1)
```

Comment:
Can we pull: `const deploymentTransaction = contractInstance.deploymentTransaction()` just like we do in the `deployProxy` function? We are extracting deployment transaction several times here.

Reply:
529b416